src/test/java/technology/dice/dicefairlink/iterators/RandomisedCyclicIteratorTest.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) 2018 - present by Dice Technology Ltd.
+ *
+ * Please see distribution for license.
+ */
+package technology.dice.dicefairlink.iterators;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+import org.assertj.core.api.Assertions;
+import org.junit.Before;
+import org.junit.Test;
+import java.util.ArrayList;
+
+public class RandomisedCyclicIteratorTest {
+
+  private static final int MIN_SIZE = 10;
+  private static final int MAX_SIZE = 1_000;
+  private static final String ELEMENT_PREFIX = "TEST_";
+
+  private int numberOfElementsToTest = -1;
+
+  @Before
+  public void setUp() {
+    numberOfElementsToTest = ThreadLocalRandom.current().nextInt(MIN_SIZE, MAX_SIZE);
+  }
+
+  @Test(expected = NoSuchElementException.class)
+  public void canCallOfFromEmptyList() {
+    RandomisedCyclicIterator cyclicIterator = RandomisedCyclicIterator.of(Collections.EMPTY_LIST);
+
+    Assertions.assertThat(cyclicIterator).isNotNull();
+
+    assertThat(cyclicIterator.hasNext()).isEqualTo(false);
+    cyclicIterator.next(); // final step throws
+  }
+
+  @Test(expected = NoSuchElementException.class)
+  public void canCallOfFromEmptySet() {
+    RandomisedCyclicIterator cyclicIterator = RandomisedCyclicIterator.of(Collections.EMPTY_SET);
+
+    Assertions.assertThat(cyclicIterator).isNotNull();
+
+    assertThat(cyclicIterator.hasNext()).isEqualTo(false);
+    cyclicIterator.next(); // final step throws
+  }
+
+  @Test
+  public void canOperateWithListOfSingleElement() {
+    final String singleElement = ELEMENT_PREFIX + "_1";
+
+    final List<String> listOfStrings = new LinkedList<>();
+    listOfStrings.add(singleElement);
+
+    RandomisedCyclicIterator<String> cyclicIterator
+        = RandomisedCyclicIterator.<String>of(listOfStrings);
+
+    assertThat(cyclicIterator).isNotNull();
+    for (int cycle = 0; cycle < numberOfElementsToTest; cycle++) {
+      assertThat(cyclicIterator.hasNext()).isEqualTo(true);
+      assertThat(cyclicIterator.next()).isEqualTo(singleElement);
+    }
+  }
+
+  @Test
+  public void canOperateWithSetOfSingleElement() {
```

Comment:
This test doesn't provide useful information - it's all covered by the previous one

Reply:
Removed.